framework/src/dm/arch/sidk_s5jt200/s5j_dm_connectivity.c

```diff
@@ -385,3 +397,201 @@ int dm_conn_unregister_linkdown_cb(conn_cb cb)
 	dmdbg("Link Down Callback UnRegistered Successfully\n");
 	return DM_ERROR_NONE;
 }
+
+static int8_t prv_dm_conn_scan_result(slsi_reason_t *reason)
+{
+	if (reason->reason_code != 0) {
+		dmdbg("Failed to scan : reason %d, locally generated : %d\n", \
+			  reason->reason_code, reason->locally_generated);
+		wifiscan_result = NULL;
+	} else {
+		WiFiFreeScanResults(&wifiscan_result);
+		WiFiGetScanResults(&wifiscan_result);
+		WiFiNetworkJoin((uint8_t *) CONFIG_DM_AP_SSID, strlen(CONFIG_DM_AP_SSID), \
+						NULL, (slsi_security_config_t *) getSecurityConfig(CONFIG_DM_AP_SECURITY, \
+								CONFIG_DM_AP_PASS, SLSI_WIFI_STATION_IF));
+	}
+	printf("scanned result\n");
+	return SLSI_STATUS_SUCCESS;
+}
+
+int dm_conn_get_scan_result(dm_scan_info_t **result)
+{
+	dm_scan_info_t *curr_record;
+	dm_scan_info_t *prev_record = NULL;
+	if (wifiscan_result == NULL) {
+		return DM_ERROR_NO_DATA;
+	}
+	/* Initialize pointer */
+	iterator = wifiscan_result;
+
+	do {
+		curr_record = (dm_scan_info_t *)malloc(sizeof(dm_scan_info_t));
+		if (curr_record == NULL) {
+			dm_conn_free_scan_result(result);
+			goto error_with_record;
```

Comment:
goto using only here, then you'd better use 'return X' instead of goto.
And, definitely it is error condition, but error_with_record returns DM_ERROR_NONE.
How do you think below?
do {
    if (curr_record == NULL) {
         return DM_ERROR_SOMETHING;
    }
 } while (iterator != NULL);
 curr_record_next = NULL;
 return DM_ERROR_NONE;